testing/integration-tests/src/client/mod.rs

```diff
@@ -582,3 +582,47 @@ async fn chainhead_unstable_unpin() {
         .await
         .is_err());
 }
+
+#[tokio::test]
+async fn partial_fee_estimate_correct() {
+    let ctx = test_context().await;
+    let api = ctx.client();
+
+    let alice = pair_signer(AccountKeyring::Alice.pair());
+    let hans = pair_signer(Sr25519Pair::generate().0);
+
+    let tx = node_runtime::tx()
+        .balances()
+        .transfer(hans.account_id().clone().into(), 1_000_000_000_000);
+
+    let signed_extrinsic = api
+        .tx()
+        .create_signed(&tx, &alice, Default::default())
+        .await
+        .unwrap();
+
+    // Method I: TransactionPaymentApi_query_info
+    let partial_fee_1 = signed_extrinsic.partial_fee_estimate().await.unwrap();
+
+    // Method II: TransactionPaymentApi_query_fee_details + calculations
+    let encoded_with_len = signed_extrinsic.encoded_with_len();
+    let InclusionFee {
+        base_fee,
+        len_fee,
+        adjusted_weight_fee,
+    } = api
+        .rpc()
+        .state_call_decoded::<FeeDetails>(
```

Comment:
We could have those other types next to/in this test; I quite like this test as a general sanity check that we're understanding how this stuff works!

Reply:
Done, good idea.